export (#16776)
pingcap/docs: tidb-cloud/serverless-export.md

> **Note:**
>
> We recommend you export to S3 or other supported storage services if you want to export a large amount of data.

<SimpleTab>

<div label="Export With CLI">

1. Export a specific table from a TiDB Serverless cluster to local.

   ```sh
   ticloud serverless export create -c <cluster-id> --databsae <database> --table <table>
   ```

2. Download the exported data after the export is succeeded.

   ```sh
   ticloud serverless export download -c <cluster-id> -e <export-id>
   ```
   
You can also use the cli with interactive mode.

   ```sh
   ticloud serverless export create
   ```

</div>

<div label="Export On Console">

Not supported yet.

</div>

</SimpleTab>

## Export to S3

You can export data directly to your own S3 bucket with the credentials.

<SimpleTab>

<div label="Export With CLI">

1. Export all data from a TiDB Serverless cluster to S3.

   ```sh
   ticloud serverless export create -c <cluster-id> --bucket-uri <bucket-uri> --access-key-id <access-key-id> --secret-access-key <secret-access-key>

>    ```

Reviewer: ````suggestion
   ```

````

From the verify job:
```
tidb-cloud/serverless-export.md: 72: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
```

---

> title: TiDB Serverless Export (Beta)
> summary: Learn how to export from TiDB Serverless.

Reviewer 2: ```suggestion
title: TiDB Serverless Export (Beta)
summary: Learn how to export data from TiDB Serverless clusters.
```

---

# TiDB Serverless Export (Beta)

> TiDB Serverless Export (Beta) is a service that allows you to export data from a TiDB Serverless cluster to local or an external storage service. You can use the exported data for backup, migration, data analysis, or other purposes.
> 
> You can also export data with `mysqldump`, `TiDB Dumpling`, or other tools. However, TiDB Serverless Export provides a more convenient and efficient way to export data from a TiDB Serverless cluster. It brings the following benefits:

Reviewer 2: ```suggestion
TiDB Serverless Export (Beta) is a service that enables you to export data from a TiDB Serverless cluster to local storage or an external storage service. You can use the exported data for backup, migration, data analysis, or other purposes.

While you can also export data using `mysqldump`, TiDB Dumpling, or other tools, TiDB Serverless Export offers a more convenient and efficient way to export data from a TiDB Serverless cluster. It brings the following benefits:
```

> - Convenience: Export service provides a simple and easy-to-use way to export data from a TiDB Serverless cluster. You don't need additional tools or resources.
> - Isolation: Export service uses separate computing resources, which is isolated from the computing resources used by your online services
> - Consistency: Export service ensures the consistency of the exported data without locking, which will not affect your online services.

Reviewer 2: ```suggestion
- Convenience: the export service provides a simple and easy-to-use way to export data from a TiDB Serverless cluster, eliminating the need for additional tools or resources.
- Isolation: the export service uses separate computing resources, ensuring isolation from the resources used by your online services.
- Consistency: the export service ensures the consistency of the exported data without causing locks, which does not affect your online services.
```

## Export to local

When you export to local, there are some limitations:

1. You can only export one table at a time.

> 2. The exported data will be expired after two days, please download the data in time.

Reviewer 2: Does it mean the exported data will be removed from the stashing area when it has been downloaded for two days?

Author: it will be removed from the stashing area after three days, but it will not allowed to download anymore after two days.

3. The exported data will be saved in the stashing area, which offers 250 GB storage space for each organization per region. If the storage space is full, you will not be able to export data to local.

> **Note:**
>

> > We recommend you export to S3 or other supported storage services if you want to export a large amount of data.

Reviewer 2: "other supported storage services" looks unclear. Currently,  besides S3, what storage services are supported for this feature?

Author: only s3 is supported now, we will support GCS (Google cloud storage) and Azure Blob Storage next Q

<SimpleTab>

<div label="Export With CLI">

1. Export a specific table from a TiDB Serverless cluster to local.

   ```sh

>    ticloud serverless export create -c <cluster-id> --databsae <database> --table <table>

Reviewer 2: ```suggestion
   ticloud serverless export create -c <cluster-id> --database <database> --table <table>
```

   ```

> 

Reviewer 2: What is the expected result of this step? Users will get an export ID?

Author: yes, it will output the export id

2. Download the exported data after the export is succeeded.

   ```sh
   ticloud serverless export download -c <cluster-id> -e <export-id>
   ```
   
You can also use the cli with interactive mode.

   ```sh
   ticloud serverless export create
   ```

</div>

<div label="Export On Console">

> Not supported yet.

Reviewer 2: When is it planned? If it will not happen in the upcoming one or two months, can we remove this tab and tell users that currently they can only use this feature with the CLI?

Author: It should be supported in the next quarter (before july)

</div>

</SimpleTab>

## Export to S3

> You can export data directly to your own S3 bucket with the credentials.

Reviewer 2: Are there any permissions required for S3? Do we need to indicate the permissions here and also provide the S3 doc link about how to configure the permissions?

<SimpleTab>

<div label="Export With CLI">

1. Export all data from a TiDB Serverless cluster to S3.

   ```sh
   ticloud serverless export create -c <cluster-id> --bucket-uri <bucket-uri> --access-key-id <access-key-id> --secret-access-key <secret-access-key>
   ```

2. Export from a TiDB Serverless cluster to S3 with SQL file.

   ```sh
   ticloud serverless export create -c <cluster-id> --bucket-uri <bucket-uri> --access-key-id <access-key-id> --secret-access-key <secret-access-key> --file-type SQL
   ```

> 

Reviewer 2: If users want to export data from a specific table to S3, they can also add `--database <database> --table <table>` to the command, right?

Author: yes

You can also use the cli with interactive mode.

   ```sh
   ticloud serverless export create
   ```

</div>

<div label="Export On Console">

Not supported yet.

</div>

</SimpleTab>

> ## Features

Reviewer 2: This section usually is provided before the usage steps. Can we move it to L16?

- Support exporting data to local and [Amazon S3](https://aws.amazon.com/s3/). Please Contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md) if your want to export to other storage services.
- Support exporting a specific database or table.
- Support exporting as CSV or SQL format.

## Pricing

> You will only be charged for a successful or canceled export.

Reviewer 2: for failed export, does users need to pay for RUs generated during the export process?

Author: need not

---
title: TiDB Serverless Export (Beta)
summary: Learn how to export from TiDB Serverless.
---

# TiDB Serverless Export (Beta)

TiDB Serverless Export (Beta) is a service that allows you to export data from a TiDB Serverless cluster to local or an external storage service. You can use the exported data for backup, migration, data analysis, or other purposes.

> You can also export data with `MySQL shell`, `TiDB Dumpling`, or other tools. However, TiDB Serverless Export provides a more convenient and efficient way to export data from a TiDB Serverless cluster. It brings the following benefits:

Reviewer: ```suggestion
You can also export data with `mysqldump`, `TiDB Dumpling`, or other tools. However, TiDB Serverless Export provides a more convenient and efficient way to export data from a TiDB Serverless cluster. It brings the following benefits:
```

Exporting with MySQL Shell isn't working:
https://github.com/pingcap/tidb/issues/24515
